src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityArcherBarbarian.java

```diff
@@ -0,0 +1,90 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.sounds.BarbarianSounds;
+import net.minecraft.entity.projectile.EntityArrow;
+import net.minecraft.entity.projectile.EntitySpectralArrow;
+import net.minecraft.entity.projectile.EntityTippedArrow;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+
+/**
+ * Implements the Archer Barbarian Entity
+ */
+public class EntityArcherBarbarian extends AbstractArcherBarbarian
+{
+
+    public static final ResourceLocation LOOT = new ResourceLocation(Constants.MOD_ID, "EntityArcherBarbarianDrops");
+
+    public EntityArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+    }
+
+    protected EntityArrow getArrow(final float p_190726_1_)
+    {
+        final ItemStack itemstack = this.getItemStackFromSlot(EntityEquipmentSlot.OFFHAND);
+
+        if (itemstack.getItem() == Items.SPECTRAL_ARROW)
+        {
+            final EntitySpectralArrow entityspectralarrow = new EntitySpectralArrow(this.world, this);
+            entityspectralarrow.setEnchantmentEffectsFromEntity(this, p_190726_1_);
+            return entityspectralarrow;
+        }
+        else
+        {
+            final EntityArrow entityarrow = super.getArrow(p_190726_1_);
+
+            if (itemstack.getItem() == Items.TIPPED_ARROW && entityarrow instanceof EntityTippedArrow)
+            {
+                ((EntityTippedArrow) entityarrow).setPotionEffect(itemstack);
+            }
+
+            return entityarrow;
+        }
+    }
+
+    @Override
+    public int getMaxSpawnedInChunk()
+    {
+        return 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)